portable_nproc.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+portable_nproc() {
+    OS="$(uname -s)"
+    if [ "$OS" = "Linux" ]; then
+        NPROCS="$(nproc --all)"
+    elif [ "$OS" = "Darwin" ] || \
+         [ "$(echo "$OS" | grep -q BSD)" = "BSD" ]; then
+        NPROCS="$(sysctl -n hw.ncpu)"
+    else
+        NPROCS="$(getconf _NPROCESSORS_ONLN)"  # glibc/coreutils fallback
```

Comment:
I wonder, can we just implement this with `nproc` on all platforms? I don't have a Mac, but it looks like freebsd has one at least: https://man.freebsd.org/cgi/man.cgi?query=nproc&sektion=1&format=html ?

or based on this: https://github.com/memkind/memkind/issues/33 it looks like maybe we want logicalcpu?

Reply:
I have a mac, will test!

Reply:
mac does have nproc

Reply:
This is just for building, so I think it's probably safe to rely on nproc here. I expect people building xcffib on macs will have homebrew installed :)

Reply:
moved to  nproc and removed the shell script